src/memory.rs

```diff
@@ -533,8 +533,8 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
     }
 
     pub fn write_primval(&mut self, dest: Pointer, val: PrimVal) -> EvalResult<'tcx, ()> {
-        if let Some(ptr) = val.try_as_ptr() {
-            return self.write_ptr(dest, ptr);
+        if let Some(alloc_id) = val.relocation {
+            return self.write_ptr(dest, Pointer::new(alloc_id, val.bits as usize));
```

Comment:
Not your mistake, just making a note for posterity: this `as usize` is wrong when emulating a 64-bit target on a 32-bit host. I have a TODO on `Pointer` for changing the `usize` to `u64`.